sdks/go/pkg/beam/io/textio/textio_test.go

```diff
@@ -40,3 +46,68 @@ func TestRead(t *testing.T) {
 	}
 
 }
+
+func TestRead(t *testing.T) {
+	f := "../../../../data/textio_test.txt"
+	p, s := beam.NewPipelineWithRoot()
+	lines := Read(s, f)
+	passert.Count(s, lines, "NumLines", 1)
+
+	if _, err := beam.Run(context.Background(), "direct", p); err != nil {
+		t.Fatalf("Failed to execute job: %v", err)
+	}
+}
+
+func TestReadAll(t *testing.T) {
+	f := "../../../../data/textio_test.txt"
+	p, s := beam.NewPipelineWithRoot()
+	files := beam.Create(s, f)
+	lines := ReadAll(s, files)
+	passert.Count(s, lines, "NumLines", 1)
+
+	if _, err := beam.Run(context.Background(), "direct", p); err != nil {
+		t.Fatalf("Failed to execute job: %v", err)
+	}
+}
+
+func TestWrite(t *testing.T) {
+	f := "../../../../data/textio_test.txt"
+	out := "text.txt"
+	p, s := beam.NewPipelineWithRoot()
+	lines := Read(s, f)
+	Write(s, out, lines)
+
+	if _, err := beam.Run(context.Background(), "direct", p); err != nil {
+		t.Fatalf("Failed to execute job: %v", err)
+	}
+
+	if _, err := os.Stat(out); errors.Is(err, os.ErrNotExist) {
+		t.Fatalf("Failed to write %v", out)
+	}
+	defer os.Remove(out)
+
+	outfileContents, _ := os.ReadFile(out)
+	infileContents, _ := os.ReadFile(f)
+	if got, want := string(outfileContents), string(infileContents); got != want {
+		t.Fatalf("Write() wrote the wrong contents. Got: %v Want: %v", got, want)
+	}
+}
+
+func TestImmediate(t *testing.T) {
+	f := "test2.txt"
+	if err := os.WriteFile(f, []byte("hello\ngo\n"), 0644); err != nil {
```

Comment:
Nit: consider using os.CreateTemp (https://pkg.go.dev/os#CreateTemp) instead of os.WriteFile for testing purposes.

Reply:
Good suggestions (here and below) - updated with both!